cmd/admin-prometheus.go

```diff
@@ -0,0 +1,202 @@
+	fatalIf(probe.NewError(e), "Unable to marshal into JSON.")
+	return string(jsonMessageBytes)
+}
+
+// StatConfig - container to hold the targets config.
+type StatConfig struct {
+	Targets []string `yaml:",flow" json:"targets"`
+}
+
+// String colorized stat config yaml.
+func (t StatConfig) String() string {
+	b, err := yaml.Marshal(t)
+	if err != nil {
+		return fmt.Sprintf("error creating config string: %s", err)
+	}
+	return console.Colorize("yaml", string(b))
+}
+
+// JSON jsonified stat config.
+func (t StatConfig) JSON() string {
+	jsonMessageBytes, e := json.MarshalIndent(t.Targets, "", " ")
+	fatalIf(probe.NewError(e), "Unable to marshal into JSON.")
+	return string(jsonMessageBytes)
+}
+
+// ScrapeConfig configures a scraping unit for Prometheus.
+type ScrapeConfig struct {
+	JobName       string       `yaml:"job_name" json:"jobName"`
+	BearerToken   string       `yaml:"bearer_token" json:"bearerToken"`
+	MetricsPath   string       `yaml:"metrics_path,omitempty" json:"metricsPath"`
+	Scheme        string       `yaml:"scheme,omitempty" json:"scheme"`
+	StaticConfigs []StatConfig `yaml:"static_configs,omitempty" json:"staticConfigs"`
+}
+
+const (
+	defaultPrometheusJWTExpiry = 100 * 365 * 24 * time.Hour
+)
+
+var defaultConfig = PrometheusConfig{
+	ScrapeConfigs: []ScrapeConfig{
+		{
+			JobName:     defaultJobName,
+			MetricsPath: defaultMetricsPath,
+			StaticConfigs: []StatConfig{
+				{
+					Targets: []string{""},
+				},
+			},
+		},
+	},
+}
+
+// checkAdminPrometheusSyntax - validate all the passed arguments
+func checkAdminPrometheusSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) != 1 {
+		cli.ShowCommandHelpAndExit(ctx, "prometheus", 1) // last argument is exit code
+	}
+}
+
+func generatePrometheusConfig(ctx *cli.Context) error {
+	// Get the alias parameter from cli
+	args := ctx.Args()
+	alias := args.Get(0)
+
+	if !isValidAlias(alias) {
+		fatalIf(errInvalidAlias(alias), "Invalid alias.")
+	}
+
+	hostConfig := mustGetHostConfig(alias)
+	if hostConfig == nil {
+		fatalIf(errInvalidAliasedURL(alias), "No such alias `"+alias+"` found.")
+	}
+
+	u, err := url.Parse(hostConfig.URL)
+	if err != nil {
+		return err
+	}
+
+	jwt := jwtgo.NewWithClaims(jwtgo.SigningMethodHS512, jwtgo.StandardClaims{
+		ExpiresAt: UTCNow().Add(defaultPrometheusJWTExpiry).Unix(),
+		Subject:   hostConfig.AccessKey,
+		Issuer:    "prometheus",
+	})
+
+	token, err := jwt.SignedString([]byte(hostConfig.SecretKey))
+	if err != nil {
+		return err
+	}
+
+	// Setting the values
+	defaultConfig.ScrapeConfigs[0].BearerToken = token
+	defaultConfig.ScrapeConfigs[0].Scheme = u.Scheme
+	defaultConfig.ScrapeConfigs[0].StaticConfigs[0].Targets[0] = u.Host
+
+	printMsg(defaultConfig)
+
+	return nil
+}
+
+// mainAdminPrometheus is the handle for "mc admin prometheus" command.
+func mainAdminPrometheus(ctx *cli.Context) error {
+
+	console.SetColor("yaml", color.New(color.FgGreen))
+
+	isGenerate := ctx.Bool("generate")
+	if !isGenerate {
```

Comment:
If the flag is always required it is never actually useful. The command can just be `mc admin prometheus myminio`, consistent with other commands.